### PR TITLE
feat: add mdx

### PR DESCRIPTION
**Pull request by alisonjoseph**
Ref #323

This PR adds support for basic mdx via `@mdx-js/loader` and `@next/mdx`

#### Changelog

**New**

- Autolink Header
- h1
- h2
- h3
- h4
- h5
- h6
- p
- ol
- ul
- li
- link
- blockquote

#### Testing / reviewing

Add a `test.mdx` file to the `services/web-app/pages` directory and see if the components listed above render correctly. (wasn't sure if we wanted to commit a test file or not for this until we have storybook up and running 🤷‍♀️ )

<details>
  <summary>test.mdx</summary>

```
# Markdown

## Text decoration

Emphasis, aka italics, with _asterisks_ or _underscores_. Strong emphasis, aka bold, with
**asterisks** or **underscores**. Combined emphasis with **asterisks and _underscores_**.

## Lists

- Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut
  labore et
  - Nested list item. Dolor sit amet consectetur adipiscing elit pellentesque habitant morbi
    tristique.
- dolore magna aliqua. Pharetra massa massa ultricies mi quis. Adipiscing enim eu turpis egestas
  pretium aenean.
- Massa eget egestas purus viverra accumsan in nisl nisi.

1. First ordered list item
2. Item with a nested item
   1. Nested list item
3. In markdown, the actual numbers don’t matter, just that it’s a number
4. **TODO: sync with design on ordered list styles, Gatsby styles don't match core Carbon**

## Links

[I’m a markdown link](https://www.carbondesignsystems.com)

[I’m a markdown link with title](https://www.carbondesignsystem.com 'Google’s Homepage')

[I’m a local link](/assets)

## Headings

# H1

## H2

### H3

#### H4

##### H5

###### H6

## Blockquote

> Without aesthetic, design is either the humdrum repetition of familiar clichés or a wild scramble
> for novelty. Without aesthetic, the computer is but a mindless speed machine, producing effects
> without substance, form without relevant content, or content without meaningful form.
>
> <cite>– Paul Rand</cite>
```

</details>

### CODE REVIEW

File: services/web-app/components/autolink-header/autolink-header.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ * Copyright IBM Corp. 2022, 2022
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { Link } from '@carbon/react/icons'
+import cx from 'classnames'
+// eslint-disable-next-line no-use-before-define
+import React from 'react'
```
[the remark this answers is not in the document]
Comment (alisonjoseph) [reply services/web-app/components/autolink-header/autolink-header.js]: getting a bunch of other errors now 😅

**Comment by alisonjoseph** [comment]
@jdharvey-ibm getting a bunch of linting errors now unrelated to any files I've touched.

**Comment by francinelucca** [comment]
I believe linting erros are due to adding ` '@typescript-eslint/no-use-before-define': 'error',` to rules in `.eslintrc.js` , do we want to enforce no usage before defining? if so we probably just need to update all those files that are not compliant and basically move everything around. I can help if that's what we want to do. @jdharvey-ibm @alisonjoseph

File: services/web-app/package.json [hunk 2]
```diff
@@ -26,8 +26,12 @@
     "url": "https://github.com/carbon-design-system/carbon-platform/issues"
   },
   "dependencies": {
+    "@mdx-js/loader": "^2.0.0",
+    "@next/mdx": "^12.1.0",
     "@carbon/pictograms-react": "^11.23.0",
     "@carbon/react": "^0.15.2",
+    "@mdx-js/loader": "^2.0.0",
+    "@next/mdx": "^12.1.0",
```
Comment (francinelucca) [inline services/web-app/package.json:34]: duplicate dependencies here 
"@mdx-js/loader": "^2.0.0",
    "@next/mdx": "^12.1.0",

**Review by francinelucca — CHANGES_REQUESTED** [review]
Minor comments, everything else LGTM
image
image
image

File: services/web-app/components/markdown/markdown.module.scss [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  /* remove hanging bullet */
+  margin-left: $spacing-05;
+}
+
+.ol {
+  /*  remove hanging bullet */
+  margin-left: $spacing-06;
+}
+
+/* TODO: pull in nested list prop diredtly from core Carbon for correct bullets */
```
[the remark this answers is not in the document]
Comment (alisonjoseph) [reply services/web-app/components/markdown/markdown.module.scss]: created issue #351

File: services/web-app/components/mdx/components.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  h2: H2,
+  h3: H3,
+  h4: H4,
+  h5: H5,
+  h6: H6,
+  p: P,
+  ol: OL,
+  li: LI,
+  ul: UL,
+  /* TODO: get nested lists working
```
[the remark this answers is not in the document]
Comment (alisonjoseph) [reply services/web-app/components/mdx/components.js]: created issue #351

**Review by francinelucca — APPROVED** [review]
🎉

**Review by jharvey10 — APPROVED** [review]

**Pull request merged**